### PR TITLE
Iasi ng bufr change

### LINKED ISSUE #815: IASI-NG BUFR mnemonics changes
The current read_iasing subroutine has mnemonics used by NESDIS.  These would not be consistent with mnemonics used by the direct broadcast sites (DBNet) or EUMETSAT.  NESDIS has agreed to use the EUMETSAT mnemonics, which will be consistent among all data providers.  The GSI subroutine read_iasing BUFR read statements need to be changed to the new standard.

### PR DESCRIPTION
**Pull request by wx20jjung**
**Description**

The mnemonics to read the IASI-NG data and the corresponding MetImage data were changed by NESDIS.  This PR is to change the read_iasing subroutine to use the new mnemonics.  
Resolves #815 
There are no dependencies with this change.  

Two links in the /src/enkf directory were also removed, as they are not necessary.
observer_nmmb.f90 -> observer_reg.f90
observer_wrf.f90 -> observer_reg.f90

**Type of change**

- [X] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**

NESDIS provided ~6 hours of proxy data containing these new mnemonics.  The OBSPROC group (Steve Stegall) generated the two dump files consistent with the global workflow. We tested the our changes on these data.

In order for others to test these changes, modifications to scripts in the global-workflow and channel entries in the satinfo file will be required.  The global-workflow script changes are in a separate (global-workflow) pull request.  Changes to the satinfo and scaninfo will be in a separate pull request (gsi/fix) once the script changes to the global-workflow are incorporated.
  
**Checklist**

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] New and existing tests pass with my changes
- [X] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by wx20jjung** [comment]
@ADCollard @DavidHuber-NOAA  @emilyhcliu would you review these changes?

**Comment by wx20jjung** [comment]
My ctests on Hera completed as expected.

Test project /scratch1/NCEPDEV/jcsda/Jim.Jung/save/ctests/update/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  497.71 sec
2/6 Test #2: rtma .............................   Passed  981.02 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  1105.12 sec
4/6 Test #6: global_enkf ......................   Passed  1149.28 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  1289.76 sec
6/6 Test #1: global_4denvar ...................   Passed  1965.64 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 1965.66 sec

One of my ctests on wcoss (global_4denvar) failed due to not having permission to use restricted data (RSTPROD).

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Changes look good.  Just one request that you also remove the two commented-out lines in `src/enkf/enkf_files.cmake` that reference `observer_nmmb.f90` and `observer_wrf.f90`.

**Comment by RussTreadon-NOAA** [comment]
**WCOSS2 ctests**
Install `develop` at 09212519 and `wx20jjung:IASI-NG_BUFR_change` at 57d3e98eb on Dogwood.  Run ctests with the following results
```
Test project /lfs/h2/emc/da/noscrub/russ.treadon/git/gsi/pr818/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  788.28 sec
2/6 Test #6: global_enkf ......................   Passed  913.43 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  1213.72 sec
4/6 Test #4: hafs_4denvar_glbens ..............   Passed  1214.60 sec
5/6 Test #2: rtma .............................   Passed  2530.94 sec
6/6 Test #1: global_4denvar ...................   Passed  2882.98 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 2883.00 sec
```

**Comment by RussTreadon-NOAA** [comment]
**Orion ctests**

Install develop at [`092151`](https://github.com/NOAA-EMC/GSI/commit/092125196a49e856ad760730edac01e398a84d8f) and `wx20jjung:IASI-NG_BUFR_change` at [`57d3398`](https://github.com/NOAA-EMC/GSI/commit/57d3e98eba2451e170fa2c751a93ed43802a475c) on Orion. Run ctests with the following results
```
Test project /work2/noaa/da/rtreadon/git/gsi/pr818/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #6: global_enkf ......................   Passed  850.53 sec
2/6 Test #3: rrfs_3denvar_rdasens .............   Passed  966.83 sec
3/6 Test #2: rtma .............................   Passed  1696.53 sec
4/6 Test #5: hafs_3denvar_hybens ..............   Passed  2968.41 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  3089.64 sec
6/6 Test #1: global_4denvar ...................   Passed  3603.42 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 3603.90 sec
```

**Comment by RussTreadon-NOAA** [comment]
**Hercules ctests**

Install `develop` at 09212519 and `wx20jjung:IASI-NG_BUFR_change` at 57d3e98eb on Hercules. Run ctests with the following results:
```
Test project /work/noaa/da/rtreadon/git/gsi/pr818/build
    Start 1: global_4denvar
    Start 2: rtma
    Start 3: rrfs_3denvar_rdasens
    Start 4: hafs_4denvar_glbens
    Start 5: hafs_3denvar_hybens
    Start 6: global_enkf
1/6 Test #3: rrfs_3denvar_rdasens .............   Passed  30555.97 sec
2/6 Test #6: global_enkf ......................   Passed  30910.00 sec
3/6 Test #5: hafs_3denvar_hybens ..............   Passed  31576.54 sec
4/6 Test #2: rtma .............................   Passed  31868.43 sec
5/6 Test #4: hafs_4denvar_glbens ..............   Passed  31943.67 sec
6/6 Test #1: global_4denvar ...................   Passed  32584.85 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) = 32584.86 sec
```

Note the extremely long ctest time - about _9 hours_.  This is due to extremely long queue times on Hercules.

**Comment by RussTreadon-NOAA** [comment]
@wx20jjung :  None of the ctests exercise the code changed in this PR.  You mention above that you tested these changes with NESDIS provided data.  I assume the test was successful.  It would be good to state this.

You hint at 
- g-w PR [#3216](https://github.com/NOAA-EMC/global-workflow/pull/3126)
- GSI-fix issue [#23](https://github.com/NOAA-EMC/GSI-fix/issues/23)

It is good to explicitly mention these as I did above so that reviewers and code managers better understand what's needed for this capability to be fully exercised.

**Review by RussTreadon-NOAA — COMMENTED** [review]
Agree with @DavidHuber-NOAA .  Please remove the commented out lines
```
#observer_nmmb.f90 -- This is soft-linked to observer_reg.f90
#observer_wrf.f90  -- This is soft-linked to observer_reg.f90
```
at the end of `src/enkf/enkf_files.cmake`.

The change to `src/gsi/read_iasing.f90` is straightforward but I have no way to confirm that the new mnemonic is correct.  

Is there EUMETSAT or NESDIS document we can link to this PR or the parent issue #815 to confirm that the `RPS` mnemonics are correct?

**Comment by wx20jjung** [comment]
@RussTreadon-NOAA   I use this page for most of my IASI info.  
https://user.eumetsat.int/resources/user-guides/metop-sg-iasi-ng-l1c-and-l1d-data-guide

**Comment by wx20jjung** [comment]
The two comment lines in src/enkf/enkf_files.cmake
#observer_nmmb.f90 -- This is soft-linked to observer_reg.f90
#observer_wrf.f90  -- This is soft-linked to observer_reg.f90

are removed.  Changes are pushed up to github in @wx20jjung  branch IASI-NG_BUFR_change

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve

**Review by DavidHuber-NOAA — APPROVED** [review]
LGTM

**Comment by RussTreadon-NOAA** [comment]
@wx20jjung :  This PR can be scheduled for merger into `develop` after we get one more peer review and approval.

**Review by ADCollard — APPROVED** [review]
Changes look good

**Comment by RussTreadon-NOAA** [comment]
Thank you @ADCollard and @DavidHuber-NOAA for your reviews and approvals.  I'll work with the GSI Handling Review team to get this PR merged into `develop`.

**Pull request merged**
